tests/lean/run/nested_inductive.lean

```diff
@@ -0,0 +1,41 @@
+
+/-! # Nested inductive types
+
+See "Recursion through lists and arrays" in https://blog.lean-lang.org/blog/2024-9-1-lean-4110/
+
+This test file exercises the `attach`/`unattach` API.
+-/
+
+namespace List
+
+inductive Tree where | node : List Tree → Tree
+
+namespace Tree
+
+def rev : Tree → Tree
+  | node ts => .node (ts.attach.reverse.map (fun ⟨t, _⟩ => t.rev))
+
+-- Note that `simp` now automatically removes the `attach`.
+@[simp] theorem rev_def (ts : List Tree) :
+    Tree.rev (.node ts) = .node (ts.reverse.map rev) := by
+  simp [Tree.rev]
+
+/-- Define `size` using a `foldl` over `attach`. -/
+def size : Tree → Nat
+  | node ts => 1 + ts.attach.foldl (fun acc ⟨t, _⟩ => acc + t.size) 0
```

Comment:
Let's maybe also test the variant where the user didn't use pattern matching but `x.val` in the definition.